apps/sla/tests/test_framework.py

```diff
@@ -458,3 +477,77 @@ def test_tracker_sla(self, is_closed):
                 assert sla_context.end == flaw.created_dt + timedelta(days=5)
             else:
                 assert not sla_context.sla
+
+        @pytest.mark.parametrize(
+            "component,excluded",
+            [
+                ("kpatch", True),
+                ("glibc", False),
+            ],
+        )
+        def test_exclusion(self, component, excluded):
+            """
+            Test that an exclusion policy placed as the first one using the 'in' condition
+            works for not applying SLA when these conditions are met
+            """
+            sla_file = """
+---
+name: Excluded components
+description: Components which do not need SLA
+conditions:
+  affect:
+    - PS component in:
+      - kpatch
+      - kudos
+      - kratos
+sla: null
+---
+name: Low
+description: SLA policy applied to low impact
+conditions:
+  affect:
+    - aggregated impact is low
+sla:
+  duration: 5
+  start: reported date
+  type: calendar days
+---
+name: Major Incident
+description: only for very serious cases
+conditions:
+  flaw:
+    - major incident state is approved
+sla:
+  duration: 2
+  start: reported date
+  type: calendar days
+"""
+            flaw = FlawFactory(
```

Comment:
If I am not overseeing something it seems to me that if the generated flaw is Major Incident the test case of no exclusion may fail on its last `assert`. Or am I wrong?

Reply:
You're right, I did not remember this was also set randomly. I changed it.